examples/filter.py

```diff
@@ -0,0 +1,48 @@
+'''
+Takes in any data from a sensor and filters it.
+'''
+
+import json
+import zmq
+
+
+def bayes_filter(prev, measured):
+    ''' Super lame Bayes filter'''
+    return (prev + measured)/2
+
+
+def main():
+    # {{{ wrap this stuff
+    context = zmq.Context()
+    subscriber = context.socket(zmq.SUB)
+    subscriber.connect("tcp://localhost:5000")
+    subscriber.setsockopt(zmq.SUBSCRIBE, b"sensor")
+
+    kill_sig = context.socket(zmq.SUB)
+    kill_sig.connect("tcp://localhost:9000")
+    kill_sig.setsockopt(zmq.SUBSCRIBE, b"EXIT")
+    # }}}
```

Comment:
Cool, merging the `kill_signal` socket into the `subscriber` one also fixed the filter process so it shuts down properly now.